third-party/realdds/src/dds-device-server.cpp

```diff
@@ -237,114 +256,140 @@ void dds_device_server::on_control_message_received()
+        handle_set_option( j, reply );
     }
-    else if( id.compare( "query-option" ) == 0 )
+    else if( id.compare( id_query_option ) == 0 )
     {
-        handle_query_option( j );
+        handle_query_option( j, reply );
     }
     else
     {
         DDS_THROW( runtime_error, "invalid control '" + id + "'" );
     }
 }
 
-void dds_device_server::handle_set_option( const nlohmann::json & j )
+
+void dds_device_server::handle_set_option( const nlohmann::json & j, nlohmann::json & reply )
 {
-    const std::string & option_name = rsutils::json::get< std::string >( j, "option-name" );
-    const std::string & owner_name = rsutils::json::get< std::string >( j, "owner-name" );
-    uint32_t counter = rsutils::json::get< uint32_t >( j, "counter" );
+    auto option_name = rsutils::json::get< std::string >( j, option_name_key );
+    std::string owner_name;  // default is empty, for a device option
+    rsutils::json::get_ex( j, owner_name_key, &owner_name );
 
     std::shared_ptr< dds_option > opt = find_option( option_name, owner_name );
-
     if( opt )
     {
-        try
-        {
-            float value = rsutils::json::get< float >( j, "value" );
+        float value = rsutils::json::get< float >( j, value_key );
+        if( _set_option_callback )
             _set_option_callback( opt, value ); //Handle setting option outside realdds
-            opt->set_value( value ); //Update option object. Do second to check if _set_option_callback did not throw
-            send_set_option_success( counter );
-        }
-        catch( std::exception e )
-        {
-            send_set_option_failure( counter, e.what() );
-        }
+        opt->set_value( value ); //Update option object. Do second to check if _set_option_callback did not throw
+        reply[value_key] = value;
     }
     else
     {
-        send_set_option_failure( counter, owner_name + " does not support option " + option_name );
+        if( owner_name.empty() )
+            owner_name = "device";
+        else
+            owner_name = "'" + owner_name + "'";
+        DDS_THROW( runtime_error, owner_name + " option '" + option_name + "' not found" );
     }
 }
 
-void dds_device_server::handle_query_option( const nlohmann::json & j )
+
+void dds_device_server::handle_query_option( const nlohmann::json & j, nlohmann::json & reply )
 {
-    const std::string & option_name = rsutils::json::get< std::string >( j, "option-name" );
-    const std::string & owner_name = rsutils::json::get< std::string >( j, "owner-name" );
-    uint32_t counter = rsutils::json::get< uint32_t >( j, "counter" );
+    auto option_name = rsutils::json::get< std::string >( j, option_name_key );
+    std::string owner_name;  // default is empty, for a device option
+    rsutils::json::get_ex( j, owner_name_key, &owner_name );
 
     std::shared_ptr< dds_option > opt = find_option( option_name, owner_name );
-
     if( opt )
     {
-        try
+        float value;
```

Comment:
I don't like uninitialized values. You can initialize by `opt->get_value()` and remove the else statement.

Reply:
Me neither, but it's not uninitialized, which is why Cppcheck doesn't complain. Your suggestion is good except it calls `get_value()` needlessly, which I don't like.

I had tried doing:
```
float value = _query_option_callback ? _query_option_callback( opt ) : opt->get_value();
if( _query_option_callback )
    // Ensure realdds option is up to date with actual value from callback
    opt->set_value( value );
```
But I don't find it more readable, and don't like the double if, either. I think it's best as-is.